examples/sprite_render/main.rs

```diff
@@ -0,0 +1,166 @@
+
+extern crate amethyst;
+#[macro_use]
+extern crate log;
+extern crate gfx;
+
+mod loader;
+mod sprite;
+
+use amethyst::core::cgmath::{Matrix4, Transform as CgTransform, Vector3};
+use amethyst::core::transform::{GlobalTransform, Transform, TransformBundle};
+use amethyst::ecs::Entity;
+use amethyst::input::InputBundle;
+use amethyst::prelude::*;
+use amethyst::renderer::{Camera, ColorMask, DisplayConfig, DrawFlat, Event, KeyboardInput,
+                         Pipeline, PosTex, Projection, RenderBundle, ScreenDimensions, Stage,
+                         VirtualKeyCode, WindowEvent};
+use amethyst::ui::{DrawUi, UiBundle};
+use gfx::preset::blend;
+
+use loader::SpriteSheetLoader;
+
+const BACKGROUND_COLOUR: [f32; 4] = [0.0, 0.0, 0.0, 0.0]; // black
+
+struct Example;
+
+impl State for Example {
+    fn on_start(&mut self, mut world: &mut World) {
+        initialise_camera(world);
+
+        let sprite_w = 32.;
+        let sprite_h = 32.;
+        let sprite_sheet_loader = SpriteSheetLoader;
+        let sprite_sheet = sprite_sheet_loader.load(
+            "texture/bat.32x32.png",
+            sprite::Metadata::new(sprite_w, sprite_h, 2, 6, false),
+            &mut world,
+        );
+
+        // Calculate offset to centre all sprites
+        //
+        // The X offset needs to be multiplied because we are drawing the sprites across the window;
+        // we don't need to multiply the Y offset because we are only drawing the sprites in 1 row.
+        let sprite_count = sprite_sheet.sprite_meshes.len();
+        let sprite_offset_x = sprite_count as f32 * sprite_w / 2.;
+        let sprite_offset_y = sprite_h / 2.;
+
+        let (width, height) = {
+            let dim = world.read_resource::<ScreenDimensions>();
+            (dim.width(), dim.height())
+        };
+        // This `Transform` moves the sprites to the middle of the window
+        let mut common_transform = Transform::default();
+        common_transform.translation = Vector3::new(
+            width / 2. - sprite_offset_x,
+            height / 2. - sprite_offset_y,
+            0.,
+        );
+
+        // Create an entity per sprite.
+        //
+        // In a real application we would probably store the `sprite_sheet` in a `Storage` instead
+        // of discarding it after this function.
+        for i in 0..sprite_count {
+            let mut sprite_transform = Transform::default();
+            sprite_transform.translation = Vector3::new(i as f32 * sprite_w, 0., 0.);
+
+            // This combines multiple `Transform`ations.
+            // You need to `use amethyst::core::cgmath::Transform`;
+            sprite_transform.concat_self(&common_transform);
+
+            world
+                .create_entity()
+                .with(sprite_sheet.sprite_meshes[i].clone())
+                .with(sprite_sheet.image.clone())
+                .with(sprite_transform)
+                // This is needed to make anything show up. I'm not sure why.
```

Comment:
It's necessary because the engine uses this to compute and store the rendered position.

Now you know why so you can remove this comment :P